File: neetcode_roadmap/binary search/153. Find Minimum in Rotated Sorted Array/solution.cpp

class Solution{
    public:
    int findMin(vector<int> &nums){
        int left =0;
        int right=nums.size()-1;
        int minimum=999999999;
        while(left<=right){
            int mid=left+(int)(right-left)/2;
            if(nums[mid]>=nums[left]){
                if(nums[mid]<minimum){
                    minimum=nums[mid];
                }else{
                    break;
                }
                left=mid;
            }
            else{
                if(nums[mid]<minimum){
                    minimum=nums[mid];
                }else{
                    break;
                }
                right=mid+1;
            }
        }
        return minimum;
            
    }
};
};

File: neetcode_roadmap/binary search/153. Find Minimum in Rotated Sorted Array/solution.py

class Solution:
    def findMin(self, nums: List[int]) -> int:
        left =0
        right=len(nums)-1
        minimum=999999999
        while(left<=right):
            mid=left+(int)(right-left)/2
            if(nums[mid]>=nums[left]):
                minimum=min(minimum,nums[mid])
                left=mid
            else:
                minimum=min(minimum,nums[mid])
                right=mid
        return minimum
